### PR TITLE
[logging] remove aInstance as input parameter to logging macros

### PR DESCRIPTION
**Pull request by abtink**

### CODE REVIEW

File: src/core/common/instance.hpp [hunk 1]
```diff
@@ -458,7 +458,7 @@ class Instance : public otInstance
     LinkRaw mLinkRaw;
 #endif // OPENTHREAD_RADIO || OPENTHREAD_ENABLE_RAW_LINK_API
 #if OPENTHREAD_CONFIG_ENABLE_DYNAMIC_LOG_LEVEL
-    otLogLevel mLogLevel;
+    static otLogLevel mLogLevel;
```
Comment (jwhui) [inline src/core/common/instance.hpp]: Is this the only static variable in OT core?

File: include/openthread/instance.h [hunk 2]
```diff
@@ -358,25 +358,22 @@ otError otInstanceErasePersistentInfo(otInstance *aInstance);
 /**
  * This function returns the current dynamic log level.
  *
- * @param[in]  aInstance A pointer to an OpenThread instance.
- *
  * @returns the currently set dynamic log level.
  *
  */
-otLogLevel otGetDynamicLogLevel(otInstance *aInstance);
+otLogLevel otGetDynamicLogLevel(void);
```
[the remark this answers is not in the document]
Comment (abtink) [reply include/openthread/instance.h]: I agree... Having them in `instance.h` does not make sense.
How about we add a new header file "openthread/logging.h" for these APIs (public ot APIs related to logging).
We already have "openthread/platform/logging.h" where we define the platform related definitions.

File: src/core/common/instance.hpp (hunk 1 above)
Comment (abtink) [reply src/core/common/instance.hpp]: I think it's the only one (not counting the `sInstanceRaw` itself). 
How about moving this into `logging.cpp`?

**Comment by abtink** [comment]
Added new changes (adding `openthread/logging.h` as a seperate commit to help with review. 
@jwhui can you please help squash before merge. Thanks

**Comment by jwhui** [comment]
@abtink, looks great, thanks.

**Pull request merged**